### ISSUE TITLE
Authelia docker container unable to authenticate with mysql docker container

### ISSUE THREAD

**Issue by kevdogg** [authelia/authelia#654]
I currently using a docker compose file to create 3 containers - mysql, redis and authelia.  

My docker compose file is the following:

```
---
version: '3.7'

networks:
  docker_net:
    ipam:
      driver: default
      config:
        - subnet: 172.28.1.0/24

services:

  redis:
    container_name: redis
    restart: unless-stopped
    image: redis:4.0-alpine
    ports:
      - 6379:6379
    networks:
      docker_net:
        ipv4_address: 172.28.1.3
    environment:

  mysql:
    container_name: mysql
    image: mysql:latest
    restart: unless-stopped
    environment:
      MYSQL_ROOT_PASSWORD: authelia_mysql_passwd
      MYSQL_DATABASE: authelia
    networks:
      docker_net:
        ipv4_address: 172.28.1.2
    ports:
      - 3306:3306
    volumes:
      - /var/data/db:/var/lib/mysql

  authelia:
    container_name: authelia
    hostname: authelia
    image: authelia/authelia:latest
    restart: unless-stopped
    networks:
      #- net
      docker_net:
        ipv4_address: 172.28.1.4
    ports:
      - 8080:8080
    volumes:
      - /etc/authelia/config/configuration.yml:/etc/authelia/configuration.yml
      - /etc/authelia/config/users.yml:/etc/authelia/users.yml
      - /var/data/authelia/db.sqlite3:/var/lib/authelia/db.sqlite3
    environment:
      NODE_TLS_REJECT_UNAUTHORIZED: '0'
```

Within the docker compose I created a network and assigned each of the containers a static IP address. I've verified that from the mysql container I can ping the redis container.  Specifically I assigned the container IPs:  (this was for the purpose of inter-container communication)
     mysql - 172.28.1.2
     redis - 172.28.1.3
     authelia - 172.28.1.4

docker confirms the networks assignments for all but the authelia container:
```
# docker network inspect authelia_docker_net
[
    {
        "Name": "authelia_docker_net",
        "Id": "c4e8103968b0266c43d9909e85757697924f1f4b73dc3efd761144908cae5bce",
        "Created": "2020-02-26T23:00:41.320660415-06:00",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.28.1.0/24"
                }
            ]
        },
        "Internal": false,
        "Attachable": true,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "8d3ffc73676af402cdd59f6fd730c67972f3b8beb74035ff5115d182edde7229": {
                "Name": "mysql",
                "EndpointID": "a9cf92e1cd3271a7c1623668a51fe312c6f4567c30b313cdb38cf01a6ffaa9e4",
                "MacAddress": "02:42:ac:1c:01:02",
                "IPv4Address": "172.28.1.2/24",
                "IPv6Address": ""
            },
            "9c820e950793896228069944a7d7757f3bc2dac2c459344916ae4c3f3ecb2edb": {
                "Name": "redis",
                "EndpointID": "087d23d1098a3ea671385eae368e8c7f144c747a3fb3ead176eeb29f5718fdc8",
                "MacAddress": "02:42:ac:1c:01:03",
                "IPv4Address": "172.28.1.3/24",
                "IPv6Address": ""
            }
        },
        "Options": {},
        "Labels": {
            "com.docker.compose.network": "docker_net",
            "com.docker.compose.project": "authelia",
            "com.docker.compose.version": "1.25.4"
        }
    }
]
```

The mysql container is created in docker compose with an external volume to save the database.
I did a docker exec -it mysql /bin/bash into the container, and added an authelia user specifically identified by the IP address of authelia:

```
mysql -u root -p

CREATE USER 'authelia'@'172.28.1.4' IDENTIFIED BY 'authelia_mysql_pass';
GRANT ALL ON authelia.* TO 'authelia'@'172.28.1.4';
FLUSH PRIVILEGES;
```

With the authelia configuration.yml file I have the following snippet:
```
storage:
  mysql:
    host: 172.28.1.2
    port: 3306
    database: authelia
    username: authelia
    password: authelia_mysql_passwd

#  local:
#    path: /var/lib/authelia/db.sqlite3
```

The authelia container isn't being created because of an error:
```
# docker logs authelia
...
...
time="2020-02-26T23:27:12-06:00" level=info msg="Logging severity set to trace"
time="2020-02-26T23:27:12-06:00" level=fatal msg="Unable to initialize SQL database: Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'IF NOT EXISTS time ON authentication_logs (time)' at line 1" stack="github.com/authelia/authelia@/internal/storage/mysql_provider.go:65 NewMySQLProvider\ngithub.com/authelia/authelia@/cmd/authelia/main.go:73               startServer\ngithub.com/authelia/authelia@/cmd/authelia/main.go:109              main.func1\ngithub.com/spf13/cobra@v0.0.5/command.go:830                        (*Command).execute\ngithub.com/spf13/cobra@v0.0.5/command.go:914                        (*Command).ExecuteC\ngithub.com/spf13/cobra@v0.0.5/command.go:864                        main\ngithub.com/spf13/cobra@v0.0.5/command.go:864                        (*Command).Execute\nruntime/proc.go:203                                                 main\nruntime/asm_amd64.s:1357                                            goexit"
...
...
```

Do I need to use a different mysql container version?
Not exactly a mysql expert.

**Comment by nightah** [comment]
Authelia doesn't work with MySQL currently, it is tested and known to work with MariaDB.
Can I suggest you try that please? We likely will support MySQL it's just not something we have got around to yet.

**Comment by clems4ever** [comment]
@nightah is right, this issue is a duplicate of https://github.com/authelia/authelia/issues/512.
@kevdogg , please try with a MariaDB or PostgreSQL instead if possible.

I'm closing this issues since it's a duplicate. Please follow up on #512.

**Comment by kevdogg** [comment]
@clems4ever 
I can verify that it worked with mariadb.

Only thing I'd add is that I'm using docker container with mariadb.  The main mariadb container (https://hub.docker.com/_/mariadb) wouldn't work for me correctly since I couldn't login.  I set the root password with the environment, but after docker exec -it ... /bin/bash and trying to login as:
mysql -u root -p
It wouldn't take the main password.  I've unfortunately seen others post about this error as well, although reports are scattered over a long time point.  I ended up using: https://hub.docker.com/r/linuxserver/mariadb which worked as expected.
